### PR TITLE
TypeScript 2.0 compilation compliance

### PR DESCRIPTION
**Pull request by nockawa**

### CODE REVIEW

File: src/Mesh/babylon.abstractMesh.ts [hunk 1]
```diff
@@ -374,7 +374,7 @@
                 this.rotation = Vector3.Zero();
             }
             var rotationQuaternion: Quaternion;
-            if (!space || space === Space.LOCAL) {
+            if (!space || (space as any) === Space.LOCAL) {
```
Comment (deltakosh) [inline src/Mesh/babylon.abstractMesh.ts]: don't understand why we need this

File: src/Tools/babylon.database.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ module BABYLON {
         private isSupported: boolean;
 
         // Handling various flavors of prefixed version of IndexedDB
-        private idbFactory = <IDBFactory> (window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB);
+        private idbFactory = <IDBFactory> (window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || (window as any).msIndexedDB);
```
Comment (deltakosh) [inline src/Tools/babylon.database.ts]: same here

File: src/Mesh/babylon.abstractMesh.ts (hunk 1 above)
Comment (nockawa) [inline src/Mesh/babylon.abstractMesh.ts]: honestly, I don't either, I've search the compilation error on google several minutes, but it just doesn't want to compile with the === operator. and there's nothing on google about it.
I also tried a const enum, it doesn't change...

File: src/Tools/babylon.database.ts (hunk 2 above)
Comment (nockawa) [inline src/Tools/babylon.database.ts]: the field is not recognized as part of the Window type....

Comment (deltakosh) [inline src/Tools/babylon.database.ts]: we have the mixin.s for that

**Comment by deltakosh** [comment]
The check is failing :(

File: src/Tools/babylon.database.ts [hunk 3]
```diff
@@ -530,7 +530,7 @@ module BABYLON {
                                 try {
                                     //backwards compatibility with ts 1.0, srcElement doesn't have an "error" according to ts 1.3
                                     if (event.srcElement['error'] && event.srcElement['error'].name === "QuotaExceededError") {
-                                        this.hasReachedQuota = true;
+                                        (this as any).hasReachedQuota = true;
```
Comment (deltakosh) [inline src/Tools/babylon.database.ts]: same here? why this?

File: src/Physics/Plugins/babylon.cannonJSPlugin.ts [hunk 4]
```diff
@@ -116,7 +116,7 @@
                                 this.removePhysicsBody(childImpostor);
                                 childImpostor.physicsBody = null;
                             }
-                            childImpostor.parent = mainImpostor;
+                            (childImpostor as any).parent = mainImpostor;
```
Comment (deltakosh) [inline src/Physics/Plugins/babylon.cannonJSPlugin.ts]: why?

File: src/Mesh/babylon.abstractMesh.ts (hunk 1 above)
Comment (deltakosh) [inline src/Mesh/babylon.abstractMesh.ts]: why not "===" ?

File: src/Tools/babylon.database.ts (hunk 3 above)
Comment (nockawa) [inline src/Tools/babylon.database.ts]: if I replace the "function" by a lambda, it does compile.

File: src/Physics/Plugins/babylon.cannonJSPlugin.ts (hunk 4 above)
Comment (nockawa) [inline src/Physics/Plugins/babylon.cannonJSPlugin.ts]: well, PhysicImpostor.parent doesn't have a setter! so unless the property setter is added at runtime, it's a bug! :)

Comment (deltakosh) [inline src/Physics/Plugins/babylon.cannonJSPlugin.ts]: This is not the way to fix it then :) I don;t want to do a fast fix
Ping @RaananW

File: src/Tools/babylon.database.ts (hunk 3 above)
Comment (deltakosh) [inline src/Tools/babylon.database.ts]: Ping @davrous

File: src/Physics/Plugins/babylon.cannonJSPlugin.ts (hunk 4 above)
Comment (RaananW) [inline src/Physics/Plugins/babylon.cannonJSPlugin.ts]: let me check :smile:

File: src/Tools/babylon.database.ts (hunk 3 above)
Comment (nockawa) [inline src/Tools/babylon.database.ts]: it compile with a lambda, otherwise we can keep a function and declare "var self = this" outside the function and use "self" instead of this and it will also compile

Comment (deltakosh) [inline src/Tools/babylon.database.ts]: There is something wrong here. We need to wait for @davrous comments

File: src/Canvas2d/babylon.canvas2d.ts [hunk 5]
```diff
@@ -1688,7 +1688,15 @@
                 this._worldSpaceNode = settings.customWorldSpaceNode;
                 this.applyCachedTexture(null, null);
             }
+
+            this.propertyChanged.add((e, st) => {
```
Comment (deltakosh) [inline src/Canvas2d/babylon.canvas2d.ts]: this seems not related to TS..do you mind doing another PR for this?

Comment (nockawa) [inline src/Canvas2d/babylon.canvas2d.ts]: well, that was the original intend of the branch....which slightly derived.
if I do a separated PR it will be on uncompiled code (since i've got TS 2.0 now). but it's a copy/paste... ok...let's try! :)

Comment (deltakosh) [inline src/Canvas2d/babylon.canvas2d.ts]: This is just because I would like to merge this now but I cannot yet merge ts2 stuff

File: src/Tools/babylon.database.ts (hunk 2 above)
Comment (davrous) [inline src/Tools/babylon.database.ts]: They have probably removed msIndexedDB from their d.ts, we have to add it to our mixin indeed rather than casting it as any.

Comment (nockawa) [inline src/Tools/babylon.database.ts]: yep, that's what I did
